managed/services/backup/backup_service.go

```diff
@@ -288,14 +288,17 @@ func (s *Service) RestoreBackup(ctx context.Context, serviceID, artifactID strin
 			jobType = models.MySQLRestoreBackupJob
 			jobData = &models.JobData{
 				MySQLRestoreBackup: &models.MySQLRestoreBackupJobData{
+					ServiceID: serviceID,
```

Comment:
Since you found this bug I think we should add some tests that now it works as expected.

Reply:
we aren't using it anywhere, for now, so I've reverted it to keep the changes within scope.